src/codec/gzip/decoder.rs

```diff
@@ -80,8 +80,20 @@ impl GzipDecoder {
 
                 State::Decoding => {
                     let prior = output.written().len();
-                    let done = inner(self, input, output)?;
+
+                    let done = inner(self, input, output).or_else(|err| match err {
+                        // we need to make an exception, if output flushed
+                        // from the decoder, to update the crc
+                        err if err.kind() == std::io::ErrorKind::Other
+                            && !output.written().is_empty() =>
+                        {
+                            Ok(false)
+                        }
+                        _ => Err(err),
+                    })?;
+
                     self.crc.update(&output.written()[prior..]);
```

Comment:
How about changing it to this to avoid swallowing error:

```suggestion
                    let res = inner(self, input, output);

                    if (output.written().len() > prior) {
                        self.crc.update(&output.written()[prior..]);
                    }
                    
                    let done = res?;
```